core/app/models/spree/order_updater.rb

```diff
@@ -98,20 +98,22 @@ def update_item_total
     end
 
     def persist_totals
-      order.update_columns(
-        payment_state: order.payment_state,
-        shipment_state: order.shipment_state,
-        item_total: order.item_total,
-        item_count: order.item_count,
-        adjustment_total: order.adjustment_total,
-        included_tax_total: order.included_tax_total,
-        additional_tax_total: order.additional_tax_total,
-        payment_total: order.payment_total,
-        shipment_total: order.shipment_total,
-        promo_total: order.promo_total,
-        total: order.total,
-        updated_at: Time.now,
+      attributes = order.slice(
+        :payment_state,
+        :shipment_state,
+        :item_count,
+        :item_total,
+        :adjustment_total,
+        :included_tax_total,
+        :additional_tax_total,
+        :payment_total,
+        :shipment_total,
+        :promo_total,
+        :total
       )
+      totals = order.class.new(attributes)
+      raise ActiveRecord::RecordInvalid, totals unless totals.valid?
+      order.update_columns(attributes.update(updated_at: Time.now))
```

Comment:
hey guys another thing to keep in mind before merging

this `valid?` call on a new Order might bring trouble (I think it could bring trouble for existing orders too), the whole idea of using `update_columns` here is to skip all rails callback chains. So calling `valid?` here might either bring unexpected results / loops and depending on the validations you have on Order it will bring performance penalties as well because they'll run when one probably wouldn't expect (for example it could trigger another db query to see if the order number is there or not or anything similar)

You think that's something worth considering? Rails `valid?` can be a lot trickier than we expect, this would get merged on master only right?